### PR TITLE
Add MaterialTags for the different Types of Tools

### PR DESCRIPTION
**Pull request by thelooter**
Adds MaterialTags for the following Types of Tools:

- Wooden Tools
- Stone Tools
- Iron Tools
- Gold Tools
- Diamond Tools
- Netherite Tools

### CODE REVIEW

**Comment by imDaniX** [comment]
Just by tags names, it's actually a bit unclear what's included into those - we also have items like shears, flint'n'steel, clock and compass, bucket.

**Comment by thelooter** [comment]
What Names would you suggest instead?

**Comment by cody-quinn** [comment]
I disagree in the names being unclear. I feel the sheers, flint n steel, click, compass and bucket fit into their own category independent of the main tiered tools.

File: patches/api/0157-Add-Material-Tags.patch [hunk 1]
```diff
@@ -777,6 +778,41 @@ index 0000000000000000000000000000000000000000..728bc33cd6081f9462ab62d2ca693557
...
++        .add(Material.GOLDEN_AXE, Material.GOLDEN_HOE, Material.GOLDEN_PICKAXE, Material.GOLDEN_SHOVEL, Material.GOLDEN_SWORD)
++        .ensureSize("GOLDEN_TOOLS", 5).lock();
++
++    /**
++     * Covers all Diamond Tools.
++     */
++    public static final MaterialSetTag DIAMOND_TOOLS = new MaterialSetTag(keyFor("diamond_tools"))
++        .add(Material.DIAMOND_AXE, Material.DIAMOND_HOE, Material.DIAMOND_PICKAXE, Material.DIAMOND_SHOVEL, Material.DIAMOND_SWORD)
++        .ensureSize("DIAMOND_TOOLS", 5).lock();
 +}
```
Comment (cody-quinn) [inline patches/api/0157-Add-Material-Tags.patch]: Where is Netherite

**Review by cody-quinn — CHANGES_REQUESTED** [review]
Seems that netherite was ignored.

Could also be cool to include armor sets?

**Comment by imDaniX** [comment]
> I disagree in the names being unclear. I feel the sheers, flint n steel, click, compass and bucket fit into their own category independent of the main tiered tools.

I mean, while bucket, compass and clock don't really fit there, shears and flint'n'steel absolutely do - both are indeed tools, made using iron, both do have durability and can be enchanted. If we don't include these in iron tools, at least is should be noted in javadocs.

**Comment by cody-quinn** [comment]
> > I disagree in the names being unclear. I feel the sheers, flint n steel, click, compass and bucket fit into their own category independent of the main tiered tools.
> 
> I mean, while bucket, compass and clock don't really fit there, shears and flint'n'steel absolutely do - both are indeed tools, made using iron, both do have durability and can be enchanted. If we don't include these in iron tools, at least is should be noted in javadocs.

I can agree with the information about them being exlucded should be included in the JavaDocs. However I still don't think they should be included in the collection itself as they are not tiered. The naming of wooden, iron, exc. makes me assume the collections are refering to the tiered tools where diamond will include all the diamond equivalents of the iron tools.

Maybe an `UNTIERED_TOOLS` or `OTHER_TOOLS` collection could be made for those?

File: patches/api/0157-Add-Material-Tags.patch (hunk 1 above)
Comment (thelooter) [reply patches/api/0157-Add-Material-Tags.patch]: Should be added now

**Review by cody-quinn — APPROVED** [review]
LGTM :+1:

**Review by NoahvdAa — APPROVED** [review]
looks good from web

**Review by Owen1212055 — APPROVED** [review]
In the future, I recommend creating PRS on other branches (not your master branch, because it makes some things difficult) 

Looks good tho

**Comment by thelooter** [comment]
> In the future, I recommend creating PRS on other branches (not your master branch, because it makes some things difficult)
> 
> Looks good tho

Yeah I usually do. Only noticed I was on master after opening the PR.

**Comment by thelooter** [comment]
Any update on this?

**Review by electronicboy — APPROVED** [review]

**Pull request merged**